coderd/database/queries/workspaces.sql

```diff
@@ -761,3 +797,6 @@ WHERE
 	-- Authorize Filter clause will be injected below in GetAuthorizedWorkspacesAndAgentsByOwnerID
 	-- @authorize_filter
 GROUP BY workspaces.id, workspaces.name, latest_build.job_status, latest_build.job_id, latest_build.transition;
+
+-- name: GetWorkspacesByTemplateID :many
+SELECT * FROM workspaces WHERE template_id = $1 AND deleted = false;
```

Comment:
We should probably make a (partial) index for this.

```sql
CREATE INDEX ON workspaces (template_id) WHERE deleted = FALSE;
```

We can do it in a follow up PR though. We don't actually have a single index on this column so it'll be good to have, especially since it's a foreign key and one would expect fk lookups to happen.

Reply:
Have added this now 👍 It's a very small change so I'm happy for it to be included.